src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNode.cs

```diff
@@ -0,0 +1,285 @@
+﻿// MUX Reference TreeViewNode.cpp, commit de78834
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class TreeViewNode : DependencyObject, ICustomPropertyProvider, IStringable
+	{
+		private WeakReference<TreeViewNode> m_parentNode;
+		private bool m_HasUnrealizedChildren;
+		private object m_itemsSource = null;
+		private ItemsSourceView m_itemsDataSource = null;
+
+		public TreeViewNode()
+		{
+			var collection = new TreeViewNodeVector();
+			collection.SetParent(this);
+			Children = collection;
+			collection.VectorChanged += ChildVectorChanged;
+		}
+
+		public TreeViewNode Parent
+		{
+			get
+			{
+				TreeViewNode parentNode = null;
+				if (m_parentNode?.TryGetTarget(out parentNode) == true)
+				{
+					return parentNode;
+				}
+				return null;				
+			}
+
+			internal set
+			{
+				if (value != null)
+				{
+					m_parentNode = new WeakReference<TreeViewNode>(value);
+				}
+				else
+				{
+					m_parentNode = null;
+				}
+
+				//A parentless node has a depth of -1, and the first level of visible nodes in
+				// the tree view will have a depth of 0 (-1 + 1);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)